From 574405659450322ce690fab758779d33ae2b5c64 Mon Sep 17 00:00:00 2001
From: Camelia Groza <camelia.groza@nxp.com>
Date: Wed, 2 May 2018 17:21:24 +0300
Subject: [PATCH 080/448] sdk: dts: ppc: add chosen node for Jumbo frame support

On FMan v2 platforms, the FMan port parameters need to be modified when
enabling Jumbo frame support. Add the necessary changes in these files.

Signed-off-by: Camelia Groza <camelia.groza@nxp.com>
---
 .../dts/fsl/qoriq-fman-0-chosen-fifo-resize.dtsi   |  103 ++++++++++++++++++++
 .../dts/fsl/qoriq-fman-1-chosen-fifo-resize.dtsi   |  103 ++++++++++++++++++++
 2 files changed, 206 insertions(+), 0 deletions(-)
 create mode 100644 arch/powerpc/boot/dts/fsl/qoriq-fman-0-chosen-fifo-resize.dtsi
 create mode 100644 arch/powerpc/boot/dts/fsl/qoriq-fman-1-chosen-fifo-resize.dtsi

diff --git a/arch/powerpc/boot/dts/fsl/qoriq-fman-0-chosen-fifo-resize.dtsi b/arch/powerpc/boot/dts/fsl/qoriq-fman-0-chosen-fifo-resize.dtsi
new file mode 100644
index 0000000..2085d9c
--- /dev/null
+++ b/arch/powerpc/boot/dts/fsl/qoriq-fman-0-chosen-fifo-resize.dtsi
@@ -0,0 +1,103 @@
+&soc {
+	chosen {
+		name = "chosen";
+		dpaa-extended-args {
+			fman0-extd-args {
+				cell-index = <0>;
+				compatible = "fsl,fman-extended-args";
+				total-fifo-size = <0x27000>;
+				fman0_oh1-extd-args {
+					cell-index = <1>;
+					compatible = "fsl,fman-port-op-extended-args";
+					fifo-size = <0x800 0x0>;
+					num-dmas = <0x1 0x1>;
+					num-tnums = <0x1 0x4>;
+				};
+				fman0_rx0-extd-args {
+					cell-index = <0>;
+					compatible = "fsl,fman-port-1g-rx-extended-args";
+					fifo-size = <0x2f00 0x2000>;
+					num-dmas = <0x1 0x1>;
+					num-tnums = <0x4 0x4>;
+				};
+				fman0_tx0-extd-args {
+					cell-index = <0>;
+					compatible = "fsl,fman-port-1g-tx-extended-args";
+					fifo-size = <0x2c00 0x0>;
+					num-dmas = <0x1 0x0>;
+					num-tnums = <0x4 0x4>;
+				};
+				fman0_rx1-extd-args {
+					cell-index = <1>;
+					compatible = "fsl,fman-port-1g-rx-extended-args";
+					fifo-size = <0x2f00 0x2000>;
+					num-dmas = <0x1 0x1>;
+					num-tnums = <0x4 0x4>;
+				};
+				fman0_tx1-extd-args {
+					cell-index = <1>;
+					compatible = "fsl,fman-port-1g-tx-extended-args";
+					fifo-size = <0x2c00 0x0>;
+					num-dmas = <0x1 0x0>;
+					num-tnums = <0x4 0x4>;
+				};
+				fman0_rx2-extd-args {
+					cell-index = <2>;
+					compatible = "fsl,fman-port-1g-rx-extended-args";
+					fifo-size = <0x2f00 0x2000>;
+					num-dmas = <0x1 0x1>;
+					num-tnums = <0x4 0x4>;
+				};
+				fman0_tx2-extd-args {
+					cell-index = <2>;
+					compatible = "fsl,fman-port-1g-tx-extended-args";
+					fifo-size = <0x2c00 0x0>;
+					num-dmas = <0x1 0x0>;
+					num-tnums = <0x4 0x4>;
+				};
+				fman0_rx3-extd-args {
+					cell-index = <3>;
+					compatible = "fsl,fman-port-1g-rx-extended-args";
+					fifo-size = <0x2f00 0x2000>;
+					num-dmas = <0x1 0x1>;
+					num-tnums = <0x4 0x4>;
+				};
+				fman0_tx3-extd-args {
+					cell-index = <3>;
+					compatible = "fsl,fman-port-1g-tx-extended-args";
+					fifo-size = <0x2c00 0x0>;
+					num-dmas = <0x1 0x0>;
+					num-tnums = <0x4 0x4>;
+				};
+				fman0_rx4-extd-args {
+					cell-index = <4>;
+					compatible = "fsl,fman-port-1g-rx-extended-args";
+					fifo-size = <0x2f00 0x2000>;
+					num-dmas = <0x1 0x1>;
+					num-tnums = <0x4 0x4>;
+				};
+				fman0_tx4-extd-args {
+					cell-index = <4>;
+					compatible = "fsl,fman-port-1g-tx-extended-args";
+					fifo-size = <0x2c00 0x0>;
+					num-dmas = <0x1 0x0>;
+					num-tnums = <0x4 0x4>;
+				};
+				fman0_rx8-extd-args {
+					cell-index = <0>;
+					compatible = "fsl,fman-port-10g-rx-extended-args";
+					fifo-size = <0x7000 0x3000>;
+					num-dmas = <0x8 0x8>;
+					num-tnums = <0x10 0x8>;
+				};
+				fman0_tx8-extd-args {
+					cell-index = <0>;
+					compatible = "fsl,fman-port-10g-tx-extended-args";
+					fifo-size = <0x4000 0x0>;
+					num-dmas = <0x8 0x0>;
+					num-tnums = <0x10 0x8>;
+				};
+			};
+		};
+	};
+};
diff --git a/arch/powerpc/boot/dts/fsl/qoriq-fman-1-chosen-fifo-resize.dtsi b/arch/powerpc/boot/dts/fsl/qoriq-fman-1-chosen-fifo-resize.dtsi
new file mode 100644
index 0000000..cdc85d2
--- /dev/null
+++ b/arch/powerpc/boot/dts/fsl/qoriq-fman-1-chosen-fifo-resize.dtsi
@@ -0,0 +1,103 @@
+&soc {
+	chosen {
+		name = "chosen";
+		dpaa-extended-args {
+			fman1-extd-args {
+				cell-index = <1>;
+				compatible = "fsl,fman-extended-args";
+				total-fifo-size = <0x27000>;
+				fman1_oh1-extd-args {
+					cell-index = <1>;
+					compatible = "fsl,fman-port-op-extended-args";
+					fifo-size = <0x800 0x0>;
+					num-dmas = <0x1 0x1>;
+					num-tnums = <0x1 0x4>;
+				};
+				fman1_rx0-extd-args {
+					cell-index = <0>;
+					compatible = "fsl,fman-port-1g-rx-extended-args";
+					fifo-size = <0x2f00 0x2000>;
+					num-dmas = <0x1 0x1>;
+					num-tnums = <0x4 0x4>;
+				};
+				fman1_tx0-extd-args {
+					cell-index = <0>;
+					compatible = "fsl,fman-port-1g-tx-extended-args";
+					fifo-size = <0x2c00 0x0>;
+					num-dmas = <0x1 0x0>;
+					num-tnums = <0x4 0x4>;
+				};
+				fman1_rx1-extd-args {
+					cell-index = <1>;
+					compatible = "fsl,fman-port-1g-rx-extended-args";
+					fifo-size = <0x2f00 0x2000>;
+					num-dmas = <0x1 0x1>;
+					num-tnums = <0x4 0x4>;
+				};
+				fman1_tx1-extd-args {
+					cell-index = <1>;
+					compatible = "fsl,fman-port-1g-tx-extended-args";
+					fifo-size = <0x2c00 0x0>;
+					num-dmas = <0x1 0x0>;
+					num-tnums = <0x4 0x4>;
+				};
+				fman1_rx2-extd-args {
+					cell-index = <2>;
+					compatible = "fsl,fman-port-1g-rx-extended-args";
+					fifo-size = <0x2f00 0x2000>;
+					num-dmas = <0x1 0x1>;
+					num-tnums = <0x4 0x4>;
+				};
+				fman1_tx2-extd-args {
+					cell-index = <2>;
+					compatible = "fsl,fman-port-1g-tx-extended-args";
+					fifo-size = <0x2c00 0x0>;
+					num-dmas = <0x1 0x0>;
+					num-tnums = <0x4 0x4>;
+				};
+				fman1_rx3-extd-args {
+					cell-index = <3>;
+					compatible = "fsl,fman-port-1g-rx-extended-args";
+					fifo-size = <0x2f00 0x2000>;
+					num-dmas = <0x1 0x1>;
+					num-tnums = <0x4 0x4>;
+				};
+				fman1_tx3-extd-args {
+					cell-index = <3>;
+					compatible = "fsl,fman-port-1g-tx-extended-args";
+					fifo-size = <0x2c00 0x0>;
+					num-dmas = <0x1 0x0>;
+					num-tnums = <0x4 0x4>;
+				};
+				fman1_rx4-extd-args {
+					cell-index = <4>;
+					compatible = "fsl,fman-port-1g-rx-extended-args";
+					fifo-size = <0x2f00 0x2000>;
+					num-dmas = <0x1 0x1>;
+					num-tnums = <0x4 0x4>;
+				};
+				fman1_tx4-extd-args {
+					cell-index = <4>;
+					compatible = "fsl,fman-port-1g-tx-extended-args";
+					fifo-size = <0x2c00 0x0>;
+					num-dmas = <0x1 0x0>;
+					num-tnums = <0x4 0x4>;
+				};
+				fman1_rx8-extd-args {
+					cell-index = <0>;
+					compatible = "fsl,fman-port-10g-rx-extended-args";
+					fifo-size = <0x7000 0x3000>;
+					num-dmas = <0x8 0x8>;
+					num-tnums = <0x10 0x8>;
+				};
+				fman1_tx8-extd-args {
+					cell-index = <0>;
+					compatible = "fsl,fman-port-10g-tx-extended-args";
+					fifo-size = <0x4000 0x0>;
+					num-dmas = <0x8 0x0>;
+					num-tnums = <0x10 0x8>;
+				};
+			};
+		};
+	};
+};
-- 
1.7.1

